### ISSUE TITLE
Provide access to entity associated with menu node

### ISSUE THREAD

**Issue by juho-jaakkola** [SnowdogApps/magento2-menu#83]
For the second time I ran into a case where I would need to access category data in the template (`menu/node_type/category.phtml`).

I can get for example category URL with `getCategoryUrl()`, but I'm not able to get the category object itself.

Would it be OK to expose the category e.g. with `\Snowdog\Menu\Block\NodeType\Category::getCategory()`?

Or perhaps it could even be called `getEntity()` so the same method could be implemented also for Product, CmsPage and CmsBlock node types?

**Comment by juho-jaakkola** [comment]
Or perhaps it might even be enough to just expose the node content. For categories, pages, etc, it would return the entity id.

In the class `\Snowdog\Menu\Block\NodeType\Category`:

```php
    public function getCategoryId($nodeId)
    {
        if (!isset($this->nodes[$nodeId])) {
            throw new \InvalidArgumentException('Invalid node identifier specified');
        }

        $node = $this->nodes[$nodeId];
        return (int) $node->getContent();
    }
```
This would be easier to implement than getting the entity object. But maybe not as useful e.g. in cases where the template needs to access some custom attribute.

**Comment by juho-jaakkola** [comment]
Ignore my previous comment. It would be *much* better to get access to the actual object instead of just the entity id. 

For example displaying the category image within a menu item would become as easy as:

```php
/** @var \Snowdog\Menu\Block\NodeType\Category $block */
$imageSrc = $block->getCategory()->getImageUrl();
```

**Comment by jochhop** [comment]
Thanks for your suggestion @juho-jaakkola an internal ticket has been created to check the issue: DEV-53115

**Comment by juho-jaakkola** [comment]
Merge request: https://github.com/SnowdogApps/magento2-menu/pull/89/

**Comment by juho-jaakkola** [comment]
Could I get some feedback about the issue and/or my merge request, so I know which direction to go to?

**Comment by jochhop** [comment]
Hi @juho-jaakkola , firstly thank you for your PR, however we started working on it internally as mentioned above, the changes have been approved and will be merged to develop soon

**Comment by juho-jaakkola** [comment]
Ah, I wasn't sure whether the internal ticket was just in the backlog, or already being worked on.

Apparently the code is not yet publicly available anywhere?

Will the `getEntity()` (or `getCategory()`?) method be implemented into the block object or the node object?

**Comment by kiber** [comment]
Changes are available in release/2.10.0
